axelor-base/src/main/java/com/axelor/apps/report/engine/EmbeddedReportSettings.java

```diff
@@ -42,7 +43,11 @@ public EmbeddedReportSettings generate() throws AxelorException {
 
       final ReportGenerator generator = Beans.get(ReportGenerator.class);
 
-      this.output = generator.generate(rptdesign, format, params, ReportingTool.getCompanyLocale());
+      if (this.locale == null) {
```

Comment:
We also need to check if there is parameter "Locale" in the params. If that's the case, you need to create a Locale with this value and use this Locale instead of ReportingTool.getCompanyLocale()